base/tps/src/org/dogtagpki/server/tps/rest/TokenService.java

```diff
@@ -326,7 +335,14 @@ protected void retrieveTokensWithVLV(
                 throw new PKIException("Token record not found");
             }
 
-            response.addEntry(createTokenData(record));
+            CMS.debug("TokenService.retrieveTokensWithVLV: record.tokenType="+
+ 		    record.getType());
+            if (authorizedProfiles.contains(record.getType())) {
+                CMS.debug("TokenService: token type allowed");
+                response.addEntry(createTokenData(record));
+	    } else {
+                CMS.debug("TokenService: token type not allowed");
```

Comment:
If it's not allowed, should the entire operation fail or just this particular record get ignored?